test/PushNotificationTests/PushNotificationTests.testdef

```diff
@@ -0,0 +1,18 @@
+{
+    "Tests": [
+        {
+            "Description": "Test unpackaged scenarios for PushNotifications",
+            "Filename": "PushNotificationTests.dll",
```

Comment:
1) I'm curious whether your new tests need to be run elevated or not?
2) It might help the reader if the owner of a test can call out anything special about a particular test right here in this .testdef file, say, the test needs to run elevated, or in a low-IL container, something like that, in an optional Notes property, or the alike. Consider starting this practice from the beginning.

Reply:
I strongly recommend against #2. Tests can specify RunAs: _and other properties_, and they can change. Code is the final authority; mentioning redundant information in docs is just a matter of time before docs are stale and misleading.

One thing that would be notable is if a test is 'destructive'. As a general rule unit/functional tests in the repo can be run on the local dev desktop. If a test would damage a dev's desktop it could be worth noting in a .testdef is it's not something easily identified by quickly eyeballing code, that `TestAll` doesn't run such tests by default and one could do it manually via `TestAll -Destructive` (hopefully in a VM and not the local desktop).

Are there other subtle test prereqs e.g. doesndidn't Power tests require another machine at some point? Still? If a test required a 2nd machine that could be wroth defining (if it's clear enough to articulate and worth as an option like 'Destructive'). Likewise if a test requires a reboot (before? after? during?) that could be noteworthy.

There's a lot of downstream opportunity here but we should rush to far toofast. First get the basic behavior working and checked in. Followed by docs. Then additional properties get interesting. My top 2 requests/desires after this PR...

1. **"TestPhase": [ "PR", "CI", "Nightly", "Release", "*" ]** -- Which phases(sp?) should run this? If not specified default = * (all). Update test infra so e.g. I can define a test that runs in the release and nightly builds but not in a PR build (I'd like to define a _subset_ of tests for that :-))
2. **"Platform": [ "+*", "-RS5*", "-19H*" ]** -- Which platform(s) should this run on? If not specified default = +* but one can include/exclude e.g. "[ "+*", "-RS5*", "-*de-DE*" ]" == include(+) all, exclude(-)RS5 (all arch, all lang, all release/debug) and exclude(-) all German images. Details TBD, goal is how do I specify a test doesn't run on certain images? e.g. How do I specify my tests only run on Win11 (do not run on Win10) because my feature only runs on Win11

Reply:
I hope there is no "destructive" tests today. It's against guidance. Best for these to not exist, but if they do, they should be called out clearly. Agreed that details of a test that might evolve over time shouldn't be in the ,testdef, but being 'destructive' feels like something good to be clear. Others may disagree, but I think of "requires elevation" as one property of a test that does not change evolve quickly. When a developer uses the -List feature on a local machine, seeing what tests require elevation might save the developer some headache when some tests fail to run locally, only because they need to run elevated.